### PR TITLE
openapi3: support patterned fields in Responses (#819)

### PR DESCRIPTION
**Pull request by mgirouard**
Implements patterned fields in `openapi3.Responses.Get` 

This seemed to be the quickest fix for #819. Open to other ideas as well!

### CODE REVIEW

File: openapi3/issue_819_test.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+              schema:
+                type: object
+                description: An error response body.
+`[1:]
+	sl := NewLoader()
+	doc, err := sl.LoadFromData([]byte(spec))
+	require.NoError(t, err)
+	err = doc.Validate(sl.Context)
+	require.NoError(t, err)
+	require.NotNil(t, doc.Paths["/v1/operation"].Get.Responses.Get(201))
```
Comment (fenollp) [inline openapi3/issue_819_test.go:35]: * Please rename this file to `issue819_...` (drop first `_`) so it follows the others.
* Please add a test that `Get`s a `nil` response. e.g. `require.Nil(t, doc.Paths["/v1/operation"].Get.Responses.Get(404))`

File: openapi3/response.go [hunk 2]
```diff
@@ -27,8 +27,21 @@ func (responses Responses) Default() *ResponseRef {
 	return responses["default"]
 }
 
+// Get returns a ResponseRef for the given status
+// If an exact match isn't initially found a patterned field is checked using
+// the first digit to determine the range (eg: 201 to 2XX)
 func (responses Responses) Get(status int) *ResponseRef {
-	return responses[strconv.FormatInt(int64(status), 10)]
+	st := strconv.FormatInt(int64(status), 10)
+	rref, ok := responses[strconv.FormatInt(int64(status), 10)]
```
Comment (fenollp) [inline openapi3/response.go:35]:
```suggestion
	rref, ok := responses[st]
```

File: openapi3/response.go [hunk 3]
```diff
@@ -27,8 +27,21 @@ func (responses Responses) Default() *ResponseRef {
...
+	if ok {
+		return rref
+	}
+	st = string(st[0]) + "XX"
```
Comment (fenollp) [inline openapi3/response.go:39]: shouldn't this try `string(st[0]) + string(st[1]) + "X"` first?

Also, isn't this code missing `len(st)` checks?

Comment (mgirouard) [reply openapi3/response.go:39]: It looks like there are only the specific ranges allowed in the spec.
> Only the following range definitions are allowed: 1XX, 2XX, 3XX, 4XX, and 5XX.

Perhaps a switch on `st` gives us the same guarantees all things considered?

Comment (fenollp) [reply openapi3/response.go:39]: yes please let's stick to the spec and `case` each branch. You may panic on unexpected data. Just mention that in the func's docs.

Comment (mgirouard) [reply openapi3/response.go:39]: Updated. Let me know if that works. I opted not to introduce a panic (even if the range is out of bounds) as the method didn't previously panic with invalid values -- perhaps this can change in a future release (although, I'd still probably prefer an error).

**Review by fenollp — APPROVED** [review]

**Comment by ffkingkeshava-lgtm** [comment]
Screenshot_20250810_214447
1k

**Comment by ffkingkeshava-lgtm** [comment]
Screenshot_20250810_214447

**Comment by ffkingkeshava-lgtm** [comment]
ALLU ARJUN MASS HERO

**Comment by ffkingkeshava-lgtm** [comment]
100k

**Comment by ffkingkeshava-lgtm** [comment]
@ffkingkeshava-lgtm
